config.c

```diff
@@ -229,11 +234,19 @@ static RRStatus processConfigParam(const char *keyword, const char *value,
             goto invalid_value;
         target->external_sharding = val;
     } else if (!strcmp(keyword, CONF_SLOT_CONFIG)) {
-        target->slot_config = RedisModule_Strdup(value);
-        if (!validSlotConfig(target->slot_config)) {
-            snprintf(errbuf, errbuflen-1, "invalid 'slot_config' value");
+        char *val = RedisModule_Strdup(value);
+        if (!validSlotConfig(val)) {
+            RedisModule_Free(val);
+            snprintf(errbuf, errbuflen - 1, "invalid 'slot_config' value");
             return RR_ERROR;
         }
+        if (target->slot_config != DEFAULT_SLOT_CONFIG) {
+            RedisModule_Free((void *) target->slot_config);
+        }
+        target->slot_config = val;
+        if (redis_raft.sharding_info) {
```

Comment:
redis_raft lives in raft thread context. So, probably need to create a req and handle config changes in raft thread.

Reply:
added all config get/set to raft thread